### PR TITLE
[Shop][Account] Changing customer password

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: features/account/customer_account/changing_password.feature [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    As a Customer
+    I want to be able to change my password
+
+    Background:
+        Given the store operates on a single channel in "France"
+        And there is a customer "Francis Underwood" identified by an email "francis@underwood.com" and a password "whitehouse"
+        And I am logged in as "francis@underwood.com"
+
+    @ui
+    Scenario: Changing my email
```
Comment (Arminek) [inline features/account/customer_account/changing_password.feature]: ``Changing my password``?

File: features/account/customer_account/changing_password_validation.feature [hunk 2]
```diff
@@ -0,0 +1,37 @@
+@customer_account
+Feature: Changing a customer password
```
Comment (Arminek) [inline features/account/customer_account/changing_password_validation.feature]: ``Customer password validation`` or something similar.

File: features/account/customer_account/changing_password_validation.feature [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        And I specify the confirmation password as "blackhouse"
+        And I try to save my changes
+        Then I should be notified that provided password is different than the current one
+
+    @ui
+    Scenario: Trying to change my password with a wrong confirmation password2
+        Given I want to change my password
+        And I specify the current password as "whitehouse"
+        And I specify the new password as "blackhouse"
+        And I specify the confirmation password as "greenhouse"
```
Comment (Arminek) [inline features/account/customer_account/changing_password_validation.feature]: ``I confirm this password`` WDYT?

File: src/Sylius/Behat/Page/Shop/Account/ChangePasswordPage.php [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    {
+        $this->getElement('confirmation')->setValue($password);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function checkValidationMessageFor($element, $message)
+    {
+        $foundedElement = $this->getFieldElement($element);
```
Comment (Arminek) [inline src/Sylius/Behat/Page/Shop/Account/ChangePasswordPage.php]: ``$foundElement``. We should change this variable in ``CreatePage``. 😛

File: src/Sylius/Behat/Page/Shop/Account/ChangePasswordPage.php [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        return $message === $foundedElement->find('css', '.form-error')->getText();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getDefinedElements()
+    {
+        return array_merge(parent::getDefinedElements(), [
+            'current password' => '#sylius_user_change_password_currentPassword',
```
Comment (Arminek) [inline src/Sylius/Behat/Page/Shop/Account/ChangePasswordPage.php]: We should do something with element naming convention.

File: features/account/customer_account/changing_password_validation.feature [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+    Background:
+        Given the store operates on a single channel in "France"
+        And there is a customer account "francis@underwood.com" identified by "whitehouse"
+        And I am logged in as "francis@underwood.com"
+
+    @ui
+    Scenario: Trying to change my password with a wrong current password
+        Given I want to change my password
+        And I specify the current password as "greenhouse"
```
Comment (lchrusciel) [inline features/account/customer_account/changing_password_validation.feature]: 😆

Comment (lchrusciel) [inline features/account/customer_account/changing_password_validation.feature]: When

File: features/account/customer_account/changing_password_validation.feature [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        And I specify the new password as "blackhouse"
+        And I confirm this password as "blackhouse"
+        And I try to save my changes
+        Then I should be notified that provided password is different than the current one
+
+    @ui
+    Scenario: Trying to change my password with a wrong confirmation password2
+        Given I want to change my password
+        And I specify the current password as "whitehouse"
```
Comment (lchrusciel) [inline features/account/customer_account/changing_password_validation.feature]: When (same for example below)

File: src/Sylius/Bundle/UserBundle/Resources/config/services.xml [hunk 8]
```diff
@@ -58,6 +58,11 @@
 
         <parameter key="sylius.user.validator.unique_registered_user.class">Sylius\Bundle\UserBundle\Validator\Constraints\RegisteredUserValidator</parameter>
         <parameter key="sylius.user.validator.customer_initializer.class">Sylius\Bundle\UserBundle\Validator\Initializer\CustomerInitializer</parameter>
+
+        <parameter key="sylius.form.type.user_change_password.date_class">Sylius\Bundle\UserBundle\Form\Model\ChangePassword</parameter>
```
Comment (lchrusciel) [inline src/Sylius/Bundle/UserBundle/Resources/config/services.xml]: date**.**class

Comment (GSadee) [inline src/Sylius/Bundle/UserBundle/Resources/config/services.xml]: `.date_class` is intended because it is a parameter to `UserChangePasswordType`

Comment (pamil) [inline src/Sylius/Bundle/UserBundle/Resources/config/services.xml]: So why wouldn't we rename that parameter as well?

Comment (lchrusciel) [inline src/Sylius/Bundle/UserBundle/Resources/config/services.xml]: Still, every class has its parameter under sylius.some_resource.class. And this parameter is injected to every resource type.

**Comment by pjedrzejewski** [comment]
Thank you Grzesiu!

**Pull request merged**